src/CMakeLists.txt

```diff
@@ -102,9 +102,7 @@ string(REPLACE "." ";" GPUARRAY_ABI_VERSION_NUMBERS ${GPUARRAY_ABI_VERSION})
 list(GET GPUARRAY_ABI_VERSION_NUMBERS 0 GPUARRAY_ABI_VERSION_MAJOR)
 list(GET GPUARRAY_ABI_VERSION_NUMBERS 1 GPUARRAY_ABI_VERSION_MINOR)
 math(EXPR GPUARRAY_ABI_NUMBER "1000*${GPUARRAY_ABI_VERSION_MAJOR} + ${GPUARRAY_ABI_VERSION_MINOR}")
-FILE(WRITE gpuarray/abi_version.h
-"\#ifndef GPUARRAY_ABI_VERSION\n\#define GPUARRAY_ABI_VERSION ${GPUARRAY_ABI_NUMBER}\n\#endif\n"
-)
+configure_file(gpuarray/abi_version.h.in gpuarray/abi_version.h @ONLY)
```

Comment:
You can prefix the file names with ${CMAKE_CURRENT_SOURCE_DIR} and it will place the file in the right place.